### PR TITLE
Fixes #15009 - Make the method boot_files_uri public

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by dLobatog** [comment]
@SkySymbol Can we see the template you tried to render?

**Comment by ghost** [comment]
Here it is  : https://github.com/theforeman/community-templates/blob/develop/autoyast/iPXE.erb

**Comment by mmoll** [comment]
The template referred doesn't use that method, it just uses a variable with a similar name (which should maybe changed then for clarity),but with an s added to it.

**Comment by dLobatog** [comment]
@mmoll It does use the method - https://github.com/theforeman/community-templates/blob/develop/autoyast/iPXE.erb#L10 and it also affects `Preseed iPXE default`. 

@SkySymbol I can reproduce the issue now :smile:  - the problem is just that `boot_files_uri` is private for OSs that don't implement that method, and public for other OSs. If you try to render that template on a host that has a Windows, Debian, Xen, CoreOS or AltLinux OS, it'll work. Making the method in OS public returns nice errors for OSs where `boot_media_uri` isn't defined or cannot be calculated.

To sum up, I'll merge if [test] Jenkins is green. Thanks @SkySymbol !

**Comment by dLobatog** [comment]
Merged as c23e57afbc0875565564218905a39d6d4d9a1df8, thanks @SkySymbol!

**Comment by mmoll** [comment]
@SkySymbol @dLobatog you are right, I wonder why this didn't show up in my and Dominic's tests... anyway it's solved now :+1:
